### PR TITLE
Blog post about 2.3.0 release

### PR DESCRIPTION
**Pull request by ABBAPOH**

### CODE REVIEW

File: content/blog/release-2.3.0.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+As [hinted](https://qbs.io/blog/release-2.2.1/) at last time, the main
+feature in this release is the use of the new Language Protocol Server for syntax highlightning.
+
+<!--more-->
+
+### Language Protocol Server
+
+We have added a Language Protocol Server to QBS which provides IDE-agnostic tooling
+(https://microsoft.github.io/language-server-protocol). On top of it, we added the "GoTo" feature
+to QtCreator - it is not possible to open source files from *.qbs files by simply pressing F2.
```
Comment (ThiemoVanEngelen) [inline content/blog/release-2.3.0.md]: Shouldn't this be "it is now possible"?

Comment (ABBAPOH) [reply content/blog/release-2.3.0.md]: nice catch thanks

File: content/blog/release-2.3.0.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+title: Qbs 2.3.0 released
+date: '2024-04-08'
+author: Christian Kandeler
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: That's not true, is it?

File: content/blog/release-2.3.0.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+---
+
+The [Qbs build tool](http://qbs.io) version 2.3.0 is available.
+
+## What's new
+
+As [hinted](https://qbs.io/blog/release-2.2.1/) at last time, the main
+feature in this release is the use of the new Language Protocol Server for syntax highlightning.
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: That's also not true ;)
We support "go to definition" in certain contexts and auto-completion. Highlighting is still provided by Qt Creator's built-in QML support.

File: content/blog/release-2.3.0.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+<!--more-->
+
+### Language Protocol Server
+
+We have added a Language Protocol Server to QBS which provides IDE-agnostic tooling
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: s/QBS/Qbs
Repeats.

File: content/blog/release-2.3.0.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+(https://microsoft.github.io/language-server-protocol). On top of it, we added the "GoTo" feature
+to QtCreator - it is now possible to open source files from *.qbs files by simply pressing F2.
+We have also added support for autocompletion of the module properties:
+
+![autocompletion](../../img/blog/release-2.3.0.image.1.png)
+
+### General
+We have added a possibility to export products to CMake via the new Exporter.cmake module. This
+allows to use libraries built with QBS directly with CMake projects without the need to use an
+intermidiate representation in form of the Exporter.pkgconfig.
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: "intermediate"

File: content/blog/release-2.3.0.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+[provider](https://doc.qt.io/qbs/module-providers.html#how-qbs-uses-module-providers) in favor of
+the [qbspkgconfig](https://doc.qt.io/qbs/qml-qbsmoduleproviders-qbspkgconfig.html) provider.
+In Qbs, module providers generate missing modules during the project configuration ("resolve")
+stage. Historically, we had only Qt and "fallback" module providers and there was no way to control
+which providers are run until we introduced "named" module providers in
+[Qbs 1.21](https://qbs.io/blog/release-1.21/). At the same time, we've added the
+`qbspkgconfig` provider that reads .pc files directly, avoiding expensive invocations of the
+`pkg-config` tool. Since then, we had two almost identical providers that work on top of .pc files.
+However, the problem with the "fallback" provider was that it works differently from other
+providers as well as suffer performance issues and has the same bugs as the `pkg-config` tool.
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: This doesn't work grammatically.

File: content/blog/release-2.3.0.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+Thus, we decided to deprecate the "fallback" provider in Qbs 2.3.0 and remove it in Qbs 2.4.0.
+For details how to use the `qbspkgconfig`, see the
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: "For details on how to use ..."

File: content/blog/release-2.3.0.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+[provider page](https://qbs.io//docs/qml-qbsmoduleproviders-qbspkgconfig/) in documentaion.
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: "in the documentation". Note the typo at the end.

File: content/blog/release-2.3.0.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+Also, please report any bugs in the `qbspkgconfig` to the
+[bug tracker](https://bugreports.qt.io/browse/QBS/).
+
+We have added a new [Tutorial](https://qbs.io//docs/tutorial/) section in the documentaion to make
+it easier to grasp core concepts and make user familiar with best practices. Also, we added a
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: "make users"

File: content/blog/release-2.3.0.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+new [example](https://github.com/qbs/qbs/blob/master/examples/exporters/qbs/imports/MyLibrary.qbs)
+on how to use Exporter modules.
+
+If a project needs to be re-resolved, we now print the reason. Also, we rewrote the wildards
+handling to track changes more accurate.
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: "accurately"

File: content/blog/release-2.3.0.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+traversing of the dependencies. Now, private dependencies (dependencies of static libraries) of
+products are not traversed more than once anymore
+([QBS-1714](https://bugreports.qt.io/browse/QBS-1714)).
+
+### Language
+* Module properties are now accessible for groups in modules
+  ([QBS-1770](https://bugreports.qt.io/browse/QBS-1770)).
+* The qbspkgconfig.mergeDependencies property was removed. This feature only existed because of
+  performance limitations which were fixed in [Qbs 2.1.0](https://qbs.io/blog/release-2.1.0/) and
+  thus was removed.
```
Comment (ckandeler) [inline content/blog/release-2.3.0.md]: I'd get rid of this last part; you already mentioned the removal in the previous sentence.

**Review by ckandeler — CHANGES_REQUESTED** [review]

**Review by ckandeler — APPROVED** [review]

**Pull request merged**